job-server/src/spark.jobserver/JobInfoActor.scala

```diff
@@ -33,18 +33,19 @@ class JobInfoActor(jobDao: JobDAO, contextSupervisor: ActorRef) extends Instrume
 
   override def wrappedReceive: Receive = {
     case GetJobStatuses(limit) =>
-      sender ! jobDao.getJobInfos(limit.get)
+      val jobInfos = Await.result(jobDao.getJobInfos(limit.get), 60 seconds)
```

Comment:
Please do not use Await, and please do not introduce another timeout which the user needs to consider.  Use callbacks:

    val originator = sender
    jobDao.getJobInfos(limit.get).foreach(originator ! _)

This gets the job done with NO timeouts and does not block the thread of the JobInfoActor, which is a terrible idea and probably the reason why the bug happened in the first place.